Add SSL support to Postgres driver

miguelenes1985 opened the pull request:
Implement SSL connection options for the Postgres driver. Add `ssl`, `ssl_key`, `ssl_cert`, `ssl_ca`, and `ssl_mode` configuration options. Update connection DSN generation to include SSL parameters when enabled. #18221

miguelenes1985 committed:
Add SSL support to Postgres driver

Implement SSL connection options for the Postgres driver. Add `ssl`, `ssl_key`, `ssl_cert`, `ssl_ca`, and `ssl_mode` configuration options. Update connection DSN generation to include SSL parameters when enabled.

markstory commented on src/Database/Driver/Postgres.php:
```diff
@@ -92,6 +97,24 @@ public function connect(): void
+            if ($this->_config['ssl_mode']) {
+                $dsn .= ';sslmode=' . $this->_config['ssl_mode'];
+            } else {
+                $dsn .= ';sslmode=' . 'allow';
```
```suggestion
                $dsn .= ';sslmode=allow';
```

These could be one string.

markstory commented on src/Database/Driver/Postgres.php:
```diff
@@ -92,6 +97,24 @@ public function connect(): void
+                $dsn .= ';sslkey=' . $this->_config['ssl_key'];
+            }
+            if ($this->_config['ssl_crt']) {
+                $dsn .= ';sslcert=' . $this->_config['ssl_crt'];
```
```suggestion
                $dsn .= ';sslcert=' . $this->_config['ssl_cert'];
```

We use `ssl_cert` for mysql connections, it would be good to have consistent conventions.

miguelenes1985 committed:
Update src/Database/Driver/Postgres.php

Co-authored-by: Mark Story <mark@mark-story.com>

miguelenes1985 committed:
Update src/Database/Driver/Postgres.php

Co-authored-by: Mark Story <mark@mark-story.com>

miguelenes1985 committed:
Fix Postgres SSL connection option name

Rename `ssl_crt` connection option to `ssl_cert` to align with PostgreSQL.
Update related tests and connection string building logic.

markstory commented:
Thank you 🎉
